### DISCUSSION TITLE
DFTB+ parallelization error

### DISCUSSION THREAD

**Discussion by achyuttt** [dftbplus/dftbplus#1504] [General]
Hello everyone,

As suggested in manual, for parallel processing, I have been using 8 processes and my k point mesh is 10*10*10 so in the groups section of parallel block, I entered 2 ( it should be integer divisor of process groups and k points) but I got the following error.
Processed input in HSD format written to 'dftb_pin.hsd'

Starting initialization...
--------------------------------------------------------------------------------
STOP Error in cart_init: cannot build a 2D cartesian grid with incompatible sizes
ScriptError: You must enter a valid path to the input file.

When I change my groups to 1, the calculation works without error.

I am new to DFTB+ and I would be happy to get my problem resolved.

Thank you!

**Comment by chuyuanchen** [comment]
Did you launch the code using mpirun or mpiexec? 
Parallel {
  Groups = 2
  UseOmpThreads = Yes
}
$ mpirun -n 2 dftb+ | tee output
Suppose you have 10 kpts, this will let each process to solve 5 kpts in parallel.

**Reply by achyuttt** [reply]
I launched with mpirun, for a system of 999 atoms, I tried with different settings. But the best setting that reduced the wallclock time significantly was 8 cpus with 8 threads for a 444 mesh. 

KPointsAndWeights = SupercellFolding {
4 0 0
0 4 0
0 0 4
0.5 0.5 0.5
  }

##!/bin/bash
##SBATCH -N 1
##SBATCH --ntasks-per-node=24
export OMP_NUM_THREADS=1
mpirun -n 2 dftb+ | tee output    
dp_dos band.out dos_total.dat

This setting has the same CPU and wall clock time while using 8 threads with 8 CPUs reduced the wall clock time by 3 times however it took 41 minutes for just 14 SCC cycles to reach a SCC Tolerance of 10E-6 for a 999 number of atoms.

I have been working on these calculations for so long but have not been able to figure out which setting best suits for the fastest calculation and also I am not sure if taking this much of long time is O.K for such a small system of ~1000 atoms.

**Comment by achyuttt** [comment]
I have the following settings for my calculations. But the time it takes is
very long for geometry optimization.
Can you suggest the best setting for the fastest calculation for the system
given below?

On Mon, Aug 5, 2024 at 3:25 PM chuyuanchen ***@***.***> wrote:

> Did you launch the code using mpirun or mpiexec?
> Parallel {
> Groups = 2
> UseOmpThreads = Yes
> }
> $ mpirun -n 2 dftb+ | tee output
> Suppose you have 10 kpts, this will let each process to solve 5 kpts in
> parallel.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dftbplus/dftbplus/discussions/1504#discussioncomment-10246471>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BFOC6BNV6XD22DVGUTMKGSTZP7GQ3AVCNFSM6AAAAABLI6N2OKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMRUGY2DOMI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Best Regards
Er. Achyut Subedi
Registered Mechanical Engineer
(IOE, TU, Nepal)
+977-9847703110
